pandas/core/common.py

```diff
@@ -1403,6 +1403,10 @@ def _fill_zeros(result, x, y, name, fill):
                 if np.isinf(fill):
                     np.putmask(result,(signs<0) & mask, -fill)
 
+                if "floordiv" in name:  # (PR 9308)
+                    nan_mask = ((y == 0) & (x == 0)).ravel()
+                    np.putmask(result, nan_mask, np.nan)
+
```

Comment:
profiling is the best as sometimes intuition is wrong :)

you might want to try reshaping explicitly first (just a guess) or taking a view

I think it's forcing a copy when one is not necessary I think

this op should be pretty fast - it's the copies that are getting in the way